docs/design-proposals/cloud-edge-communication/cloud-edge-communication-framework.md

```diff
@@ -11,18 +11,255 @@ Because the cloud and edge is located in the network boundary of different secur
 However, there may be many problems when VPN is used to directly connect the components/services of different security levels:
 
 * It is too risky to directly expose the central cloud management network/IP to users
-
 * IP needs to be planned globally in advance to ensure that the management components/service IP of the central cloud and each edge site do not conflict
-
 * Network changes can be difficult
 
 This proposal aims to outline a design for cloud-edge communication framework/service to solve the above problems
 
 ## Goals
 
 * Support communication access capability of cloud-edge basic management components
-
 * Support independent IP addressing and IP address convergence of edge management components/services
-
 * Minimize attack surface, zero trust security
 
+## Proposal
+
+The following two solutions do not conflict with each other. The former can converge IP addresses to resolve IP address conflicts. However, the communication needs to be first sent to the gateway and then to the destination service. The latter cannot converge IP addresses but can resolve IP address conflicts. The latter provides end-to-end communication.
+
+### Proposal A
+
+The communication of cloud edge components/services is realized based on L4/L7 proxy, which is similar to kubedge's cloud-edge message channel.
+
+Cloudgateway and edgegateway are introduced into the central cloud and each edge site respectively. The edge site edgegateway supports connecting to the cloud gateway of the central cloud when it is started. The service communication between cloud and edge needs to go through these two gateways before arriving. Service access interfaces are provided on these two gateways to receive requests and route requests to target services.
```

Comment:
what does "before arriving" mean?

"Service access interfaces are provided on these two gateways to receive requests and route requests to target services." --> these two gateways provide service access interface (API?) that receives requests and routes requests to target services.

Reply:
"before arriving" will be deleted, thx
These two gateways provide API that receives requests and routes requests to target services.